### DISCUSSION TITLE
How to source wezterm config file after modify?

### DISCUSSION THREAD

**Discussion by owenstake** [wezterm/wezterm#5120] [Q&A]
### What Operating System(s) are you running on?

Windows

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

How to source wezterm config file after modify?

**Comment by wez** [comment]
It will usually happen automatically, but you can use the https://wezfurlong.org/wezterm/config/lua/keyassignment/ReloadConfiguration.html key assignment which is CTRL-SHIFT-R in the default key assignments.  Or you can use CTRL-SHIFT-P to open the command palette and type "reload" to select the reload option from there.

**Reply by MrDowntempo** [reply]
> It will usually happen automatically, but you can use the https://wezfurlong.org/wezterm/config/lua/keyassignment/ReloadConfiguration.html key assignment which is CTRL-SHIFT-R in the default key assignments. Or you can use CTRL-SHIFT-P to open the command palette and type "reload" to select the reload option from there.

Is it possible to do this via the command line or from a bash script? I've got a script that changes color schemes, and need to tell Wezterm to reload when I change the toml file in ~/.config/wezterm/colors

I was hoping something like ```wezterm cli reload``` would be possible. 

in the meantime I'm just using:
```touch ~/.config/wezterm/wezterm.lua```

**Reply by wez** [reply]
Take a look at https://wezterm.org/config/lua/wezterm/add_to_config_reload_watch_list.html
I'd suggest using https://wezterm.org/config/lua/wezterm/glob.html to find the files in your colors directory and add them to the reload list

**Reply by MrDowntempo** [reply]
> Take a look at https://wezterm.org/config/lua/wezterm/add_to_config_reload_watch_list.html I'd suggest using https://wezterm.org/config/lua/wezterm/glob.html to find the files in your colors directory and add them to the reload list

Ah, ok. So no way to force a reload from the command line? I'm not against using something like
```wezterm.add_to_config_reload_watch_list(wezterm.config_dir .. '/colors')``` but it might be tough adding that to the a wezterm.lua, making sure it's in a reasonable place, and properly indented/formatted. This isn't a script just for me, but part of a larger project for other users (noctalia-shell). I think the touch method might be the cleanest for this case.
